Userland/Libraries/LibJS/Runtime/Temporal/Duration.cpp

```diff
@@ -760,7 +760,8 @@ ThrowCompletionOr<RoundedDuration> round_duration(GlobalObject& global_object, d
         auto result = TRY(nanoseconds_to_days(global_object, *nanoseconds_bigint, intermediate));
 
         // e. Set days to days + result.[[Days]] + result.[[Nanoseconds]] / result.[[DayLength]].
-        days += result.days + result.nanoseconds.cell()->big_integer().divided_by(Crypto::UnsignedBigInteger::create_from((u64)result.day_length)).quotient.to_double();
+        // FIXME: Can we have bigint division with floating point remainder somehow?
+        days += result.days + result.nanoseconds.cell()->big_integer().to_double() / result.day_length;
```

Comment:
can we instead keep it as is, but add an additional term that divides the remainder by the day length? it will be even uglier but it should work in more cases as far as I understand it

Reply:
Sure - like this?

Reply:
Yep, still a bit fugly due to the double to u64 cast issue, but that's another problem for another day